kloud: escape variables in comments / stub payload_ variables

rjeczalik opened the pull request:
This PR:

- escapes variables from commented-out lines in user\_data
- stubs ${var.payload_*} during plan request, so no need to send blanks explicitely

rjeczalik committed:
stack/provider: add EscapeDeadVariables function

rjeczalik committed:
kloud: escape variables in commented lines of user_data

rjeczalik committed:
kloud: stub payload_ variables on plan request

rjeczalik commented:
/cc @gokmen

cihangir commented on go/src/koding/kites/kloud/stack/provider/variable.go:
```diff
@@ -67,3 +81,40 @@ func ReplaceVariables(s string, vars []Variable, blank string) string {
+
+	if eofNL {
+		return buf.String()
+	}
```
here is something wrong with the logic. if it ends with new line, which is also true for the following case, we dont care about it and return the whole buffer, but if it doesnt have, we apply "TrimRight". In any case "Fprintln" will add new lines. What about removing this logic and using "TrimSpace"? Or am i missing something?

rjeczalik replied:
Not exactly:

If original value ended with a new line - we preserve it. If it doesn't - we trim it, because as you observed - we always write a new line for each of the scanned lines.

rjeczalik replied:
If we did not trim it, the following case would need to have a new line at the end:

https://github.com/koding/koding/pull/10453/files#diff-837eac487945eef9374603db1a552ff6R111

Since the function is only about escaping, it could be unexpected if it also appended a new-line.

cihangir replied:
> Since the function is only about escaping, it could be unexpected if it also appended a new-line.

you got me

rjeczalik committed:
klient/remote: increase timeout for pingtracker test

Improves the following flake:

    --- FAIL: TestPing (0.13s)
      pingtracker_test.go:353: Expected the new status time to me from a X ms ago. Wanted 5ms, Got 7ms

rjeczalik committed:
klient/remote: increase grace period for machine test

Improves the following flake:

    * /pipeline/build/go/src/koding/klient/remote/machine/machine_test.go
        Line 183:
        Expected '2017-01-30 15:38:41.163780514 +0000 UTC' to happen between
       '2017-01-30 15:38:41.136014015 +0000 UTC' and '2017-01-30 15:38:41.156014015 +0000 UTC'
       (it happened '7.766499ms' outside threshold)!

rjeczalik committed:
klient/remote: disable old tests

The remote package is going to be removed.

gokmen commented:
So, should we take in?

rjeczalik commented:
If @cihangir is ok with it, then 🚢 .

rjeczalik commented:
Merging for now. If there's anything more, I'll address in next PR.

cihangir commented:
LGTM
